nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/StopDomainCommand.java

```diff
@@ -140,34 +145,56 @@ protected int executeCommand()
          * so even if the password is wrong we don't want any more
          * prompting here.
          */
-
         doCommand();
         return 0;
     }
 
     /**
-     * Print message and return exit code when
-     * we detect that the DAS is not running.
+     * Print message and return exit code when we detect that the DAS is not running.
+     *
+     * @return Success in all cases
+     * @throws org.glassfish.api.admin.CommandException
      */
     protected int dasNotRunning() throws CommandException {
         if (kill) {
-            if (isLocal())
+            if (isLocal()) {
                 return kill();
-            else // remote.  We can NOT kill and we can't ask it to kill itself.
+            } else {
+                // remote.  We can NOT kill and we can't ask it to kill itself.
                 throw new CommandException(Strings.get("StopDomain.dasNotRunningRemotely"));
+            }
         }
 
         // by definition this is not an error
         // https://glassfish.dev.java.net/issues/show_bug.cgi?id=8387
-        if (isLocal())
-            logger.warning(Strings.get("StopDomain.dasNotRunning", getDomainRootDir()));
-        else
+        if (isLocal()) {
+            ListDomainsCommand listDomains = serviceLocator.getService(ListDomainsCommand.class);
+            String runningDomains = "";
+            try {
+                Map<String, Boolean> domains = listDomains.getDomains();
+                for (String domain : domains.keySet()) {
+                    if (domains.get(domain)) {
+                        runningDomains += "\n" + domain;
+                    }
+                }
+            } catch(Exception e) {        
+            }
+            runningDomains = 
+                    (runningDomains.length() < 1) 
+                    ? "\nNo domains are currently running." 
+                    : "\nPlease specify one of the currently running domains:" + runningDomains;
+            logger.warning(Strings.get("StopDomain.dasNotRunning", getDomainRootDir(), runningDomains));
+        } else {
             logger.warning(Strings.get("StopDomain.dasNotRunningRemotely"));
-        return 0;
+        }
+        // If it has gotten this far, the domain has failed to be stopped so the command has failed
+        return 1;
```

Comment:
Why have you changed the return value for the command?

Reply:
At this point the command has failed to find a valid running instance that matches the input, so it would make more sense to have the command report it as a failure.